File: source/frontend/src/lib/dto/response/aggregate/SystemElementAggregateResponseDto.ts
import {ConstraintResponseDto, fromConstraintsResponseDto} from "$lib/dto/response/entity/ConstraintResponseDto";
import {SystemElementAggregate} from "$lib/domain/aggregate/SystemElementAggregate.svelte";

export class SystemElementAggregateResponseDto {
    id : string = "";
    title : string = "";
    constraints : ConstraintResponseDto[] = [] ;

    constructor() {}
}

export function fromSystemElementAggregatesResponseDto(systemElementAggregatesResponseDto: SystemElementAggregateResponseDto[]) : SystemElementAggregate[] {
    return systemElementAggregatesResponseDto.map(fromSystemElementAggregateResponseDto);
}

export function fromSystemElementAggregateResponseDto(systemElementAggregateResponseDto : SystemElementAggregateResponseDto) : SystemElementAggregate {
    const systemElementAggregate : SystemElementAggregate = SystemElementAggregate.create();

    systemElementAggregate.id = systemElementAggregateResponseDto.id;
    systemElementAggregate.title = systemElementAggregateResponseDto.title;

    systemElementAggregate.constraints = fromConstraintsResponseDto(systemElementAggregateResponseDto.constraints);

    return systemElementAggregate;
}

File: source/backend/src/main/java/ch/unisg/backend/controller/http/dto/response/SystemElementAggregateResponseDto.java
package ch.unisg.backend.controller.http.dto.response;

import ch.unisg.backend.controller.http.dto.response.ar.ConstraintResponseDto;
import ch.unisg.backend.core.domain.aggregate.SystemElementAggregate;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public class SystemElementAggregateResponseDto {

    public static HashMap<String, Object> toJson(SystemElementAggregate systemElementAggregate) {
        HashMap<String, Object> systemElementAggregateDto = new HashMap<>();

        systemElementAggregateDto.put("id", systemElementAggregate.getSystemElement().getId().toString());
        systemElementAggregateDto.put("title", systemElementAggregate.getSystemElement().getTitle());

        List<HashMap<String, Object>> constraintsDto = new ArrayList<>();

        systemElementAggregate.getConstraintList().forEach(constraint -> {
            constraintsDto.add(ConstraintResponseDto.create(constraint));
        });

        systemElementAggregateDto.put("constraints", constraintsDto);

        return systemElementAggregateDto;
    }
}
